### PR TITLE
Update description for initializer in IR and Loop in operator

### PR DESCRIPTION
**Pull request by jcwchen**
**Description**
Add more descriptions in documents:
1. Specify that `.initializer` is not in topological order but `.node` is
2. Add a description: Subgraph input/output matching is based on order not name.

**Motivation and Context**
1. Users are confused about why `initializer` does not follow topological order https://github.com/onnx/onnx/issues/2860
2. Users thought the input/output of subgraph (produced by loop node) is matching by the name. https://github.com/onnx/onnx/issues/2498

### CODE REVIEW

**Comment by jcwchen** [comment]
Please help me to review this @gramalingam @linkerzhang 
Thank you!

**Comment by jcwchen** [comment]
@gramalingam @linkerzhang please help me to review this.
Thank you!

File: onnx/defs/controlflow/defs.cc [hunk 1]
```diff
@@ -556,6 +556,8 @@ Frontends should emit multi-layer RNNs as a series of While operators (with
 time being the inner looping dimension), with each successive layer consuming
 the scan_outputs from the previous layer, possibly going through several
 point-wise operators (e.g. dropout, residual connections, linear layer).
+
+The input/output of subgrpah (produced by loop node) matching is based on order instead of name. The implementation will figure out the names based on this order.
```
Comment (linkerzhang) [inline onnx/defs/controlflow/defs.cc:560]: typo: subgrpah  -> subgraph.

File: docs/IR.md [hunk 2]
```diff
@@ -178,8 +178,8 @@ Graphs have the following properties:
 |Name|Type|Description|
 |---|---|---|
 name|string|The name of the model graph.
-node|Node[]|A list of nodes, forming a partially ordered computation graph based on input/output data dependencies.
-initializer|Tensor[]|A list of named tensor values. When an initializer has the same name as a graph input, it specifies a default value for that input. When an initializer has a name different from all graph inputs, it specifies a constant value.
+node|Node[]|A list of nodes, forming a partially ordered computation graph based on input/output data dependencies. It is in topological order.
+initializer|Tensor[]|A list of named tensor values. When an initializer has the same name as a graph input, it specifies a default value for that input. When an initializer has a name different from all graph inputs, it specifies a constant value. It is not in topological order.
```
Comment (linkerzhang) [inline docs/IR.md:182]: suggestion: it is not in topological order --> it may not be in topological order.

**Comment by jcwchen** [comment]
Updated. Thank you @linkerzhang!

File: docs/IR.md [hunk 3]
```diff
@@ -178,8 +178,8 @@ Graphs have the following properties:
 |Name|Type|Description|
 |---|---|---|
 name|string|The name of the model graph.
-node|Node[]|A list of nodes, forming a partially ordered computation graph based on input/output data dependencies.
-initializer|Tensor[]|A list of named tensor values. When an initializer has the same name as a graph input, it specifies a default value for that input. When an initializer has a name different from all graph inputs, it specifies a constant value.
+node|Node[]|A list of nodes, forming a partially ordered computation graph based on input/output data dependencies. It is in topological order.
+initializer|Tensor[]|A list of named tensor values. When an initializer has the same name as a graph input, it specifies a default value for that input. When an initializer has a name different from all graph inputs, it specifies a constant value. It may not be not in topological order.
```
Comment (linkerzhang) [inline docs/IR.md:182]: It may not be not in topological order. 

Remove the redundant "not" :).

Comment (jcwchen) [reply docs/IR.md]: Thanks!

**Pull request merged**